oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/CommitBuilderTest.java

```diff
@@ -150,6 +150,18 @@ public void addNodeState() {
         assertNotNull(op);
     }
 
+    @Test(expected = DocumentStoreException.class)
+    public void addLongNameNodeState() {
+        Path path = Path.fromString(DocumentMK.LONG_PATH);
+        CommitBuilder builder = new CommitBuilder(ns, null);
+        builder.addNode(path);
+        Commit c = builder.build(ns.newRevision());
+        UpdateOp up = c.getUpdateOperationForNode(path);
+        UpdateOp.Operation op = up.getChanges().get(
+                new UpdateOp.Key("_deleted", c.getRevision()));
+        assertNotNull(op);
+    }
```

Comment:
builder.addNode is expected to throw an exception here - what about removing the code below and replacing it with
```suggestion
        fail("should not reach this point");
    }
```